flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/HsSpillingInfoProvider.java

```diff
@@ -79,4 +79,24 @@ enum ConsumeStatus {
         /** The buffer is either consumed or not consumed. */
         ALL
     }
+
+    /** This class represents a pair of {@link ConsumeStatus} and consumer id. */
+    class ConsumeStatusWithId {
+        public static final ConsumeStatusWithId ALL_CONSUME_STATUS =
+                new ConsumeStatusWithId(ConsumeStatus.ALL, -1);
+
+        ConsumeStatus status;
+
+        int consumerId;
```

Comment:
I think we need a dedicated class for the consumer id, where we can define the special values such as `ANY` and `SINGLE_CONSUMER`.